### PR TITLE
issue37 Replace CDI producer for transactions with Weld TransactionSe…

### PR DESCRIPTION
**Pull request by mmusgrov**
…rvices

#37

@manovotn Thanks for letting me know that I needed the Weld JTA module to make this work.

### CODE REVIEW

File: tests/src/test/java/io/smallrye/concurrency/test/JTATest.java [hunk 1]
```diff
@@ -334,4 +334,40 @@ public void testTransactionWithUT() throws Exception {
...
+    public void testTransactionWithSuspend() throws Exception {
+        ManagedExecutor executor = ManagedExecutor.builder()
+                .maxAsync(2)
+                .propagated(ThreadContext.TRANSACTION)
+                .cleared(ThreadContext.ALL_REMAINING)
+                .build();
+
+        try (WeldContainer container = weld.initialize()) {
+            // lookup the transaction manager and the test service
+            UserTransaction ut = container.select(UserTransaction.class).get();
```
Comment (manovotn) [inline tests/src/test/java/io/smallrye/concurrency/test/JTATest.java]: @mmusgrov I am surprised you don't need Weld JTA artifact here (as in `/tests` module` as well?

Comment (mmusgrov) [reply tests/src/test/java/io/smallrye/concurrency/test/JTATest.java]: I agree but I did run all the tests locally with success (I checked out commit ddb2e2c of the upstream MP-C branch - the one before the config stuff was removed).

Comment (mmusgrov) [reply tests/src/test/java/io/smallrye/concurrency/test/JTATest.java]:
``` bash
mvn test -f tests/pom.xm
[INFO] Tests run: 25, Failures: 0, Errors: 0, Skipped: 0
```

Comment (mmusgrov) [reply tests/src/test/java/io/smallrye/concurrency/test/JTATest.java]: I am debugging where the test is getting UserTransaction from ...

Comment (mmusgrov) [reply tests/src/test/java/io/smallrye/concurrency/test/JTATest.java]: So well spotted @manovotn, I had a second CDI producer for UserTransaction in the test module so I removed that.
I also moved the Weld TransactionServices implementation to the `smallrye-concurrency-jta` artefact and created a test jar for it. Both the tck and test modules now depend on this new test jar.

Comment (manovotn) [reply tests/src/test/java/io/smallrye/concurrency/test/JTATest.java]: Good job!

**Comment by mmusgrov** [comment]
@FroMage this PR has an ack from Matej so if you are happy with it could you merge it for us.

**Comment by manovotn** [comment]
@mmusgrov I can do that as well ;-)

**Pull request merged**